lib/active_record/cursor_paginator.rb

```diff
@@ -252,10 +255,39 @@ def memoize(key)
       def build_filter_query(sorted_relation, op, current_field, prev_fields)
         relation = sorted_relation
         prev_fields.each do |col, val|
-          relation = relation.where(relation.arel_table[col].send(:eq, val))
+          col = @aliases[col] if @aliases.has_key? col
+          relation = relation.where("#{col} = ?", val)
         end
         col, val = current_field
-        relation.where(relation.arel_table[col].send(op, val))
+        col = @aliases[col] if @aliases.has_key? col
+        relation.where("#{col} #{op} ?", val)
+      end
+
+      # parse aliases from select values
+      #
+      # @return [Hash]
+      def parse_aliases
+        aliases = {}
+        @relation.select_values.each do |select_value|
+          next unless select_value.is_a? String
+
+          select_value.split(',').each do |expr|
+            expr.strip!
+            # "value [AS|as] alias" という形式に対応する
+            # value: spaceがあってもOK. 最小マッチングのため '?' をつける
+            # 'as' は使わないのでキャプチャしない
+            match = expr.match(/^(?<value>.+?)\s+(?:[Aa][Ss]\s+)?(?<alias>\S+)$/)
```

Comment:
case insensitive にすれば十分な気も

```suggestion
            match = expr.match(/^(?<value>.+?)\s+(?:as\s+)?(?<alias>\S+)$/i)
```

IMOです